src/inner_product_round.rs

```diff
@@ -167,15 +175,13 @@ where
             return Ok(());
         };
 
-        n /= 2; // Rounds towards zero, truncating any fractional part
-        let a1 = &self.ai[..n];
-        let a2 = &self.ai[n..];
-        let b1 = &self.bi[..n];
-        let b2 = &self.bi[n..];
-        let gi_base_lo = &self.gi_base[..n];
-        let gi_base_hi = &self.gi_base[n..];
-        let hi_base_lo = &self.hi_base[..n];
-        let hi_base_hi = &self.hi_base[n..];
+        // Split vectors in half; since `n` is always a nontrivial power of two, this is well defined
+        n /= 2;
+        let (a1, a2) = self.ai.split_at(n);
+        let (b1, b2) = self.bi.split_at(n);
+        let (gi_base_lo, gi_base_hi) = self.gi_base.split_at(n);
+        let (hi_base_lo, hi_base_hi) = self.hi_base.split_at(n);
+
```

Comment:
I do not see this as an improvement, especially looking at what goes on behind `pub const fn split_at(&self, mid: usize)` and the fact that it will panic if something is amiss, just the same as our code.

With the addition of the new check/verify we have these:
1. `let mut n = self.gi_base.len();`
2. `(self.hi_base.len() == n && self.ai.len() == n && self.bi.len() == n && self.y_powers.len() >= n + 2)`

That means the code as we have it is well protected and will never panic. That can even go as a comment above the slicing.

We can also reduce the lines of code if that is better:
```
let (a1, a2) = (&self.ai[..n], &self.ai[n..]);
etc.
```

Reply:
Agreed that the panic condition doesn't change. The reason for the change was to make the intent (splitting the vectors) more clear, as well as to limit the possibility for error by only specifying the split index once per vector. The function `Vec::split_at` seemed perfect for this use case.